modules/media_event_dispatcher/tests/src/Unit/Media/MediaSourceInfoAlterEventTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\Tests\media_event_dispatcher\Unit\Media;
+
+use Drupal;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\hook_event_dispatcher\HookEventDispatcherInterface;
+use Drupal\media_event_dispatcher\Event\Media\MediaSourceInfoAlterEvent;
+use Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy;
+use Drupal\Tests\UnitTestCase;
+use function media_event_dispatcher_media_source_info_alter;
+
+/**
+ * Class MediaSourceInfoAlterEventTest.
+ *
+ * @package Drupal\Tests\media_event_dispatcher\Unit\Media
+ *
+ * @group media_event_dispatcher
+ */
+class MediaSourceInfoAlterEventTest extends UnitTestCase {
+
+  /**
+   * The manager.
+   *
+   * @var \Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy
+   */
+  private $manager;
+
+  /**
+   * Sets up the test.
+   */
+  public function setUp() {
+    $builder = new ContainerBuilder();
+    $this->manager = new HookEventDispatcherManagerSpy();
+    $builder->set('hook_event_dispatcher.manager', $this->manager);
+    $builder->compile();
+    Drupal::setContainer($builder);
+  }
+
+  /**
+   * MediaSourceInfoAlterEvent sources array alter test.
+   *
+   * This tests altering the media source plugin definitions array.
+   */
+  public function testMediaSourceInfoAlter() {
+    $sources = $expectedSources = [
+      'image' => [
+        'id' => 'image',
+        'label' => new TranslatableMarkup('Image'),
+        'description' => new TranslatableMarkup('Use local images for reusable media.'),
+        'class' => 'Drupal\media\Plugin\media\Source\Image',
+        'allowed_field_types' => ['image'],
+      ],
+    ];
+
+    $this->manager->setEventCallbacks([
+      HookEventDispatcherInterface::MEDIA_SOURCE_INFO_ALTER => function (
```

Comment:
I always make this function an `static function` as that is slightly faster.

Reply:
I didn't even realize it was possible to use a static function as a closure. I just learned something new. :D

Reply:
Yep, your welcome 🙂